pkg/controllers/clusterresourcebindingwatcher/watcher_integration_test.go

```diff
@@ -0,0 +1,173 @@
+			crb.Spec.State = fleetv1beta1.BindingStateBound
+			Expect(k8sClient.Update(ctx, crb)).Should(Succeed(), "failed to update cluster resource binding")
+
+			By("By checking placement controller queue")
+			eventuallyCheckPlacementControllerQueue(crb.GetLabels()[fleetv1beta1.CRPTrackingLabel], "placementController should receive the cluster resource placement name")
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - RolloutStarted", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingRolloutStarted)
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - Overridden", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingOverridden)
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - WorkCreated", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingWorkCreated)
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - Applied", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingApplied)
+		})
+
+		It("Should enqueue the clusterResourcePlacement name for reconciling, when clusterResourceBinding status changes - Available", func() {
+			updateClusterResourceBindingStatusWithCondition(fleetv1beta1.ResourceBindingAvailable)
+		})
+	})
+})
+
+func clusterResourceBindingForTest() *fleetv1beta1.ClusterResourceBinding {
+	return &fleetv1beta1.ClusterResourceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   testCRBName,
+			Labels: map[string]string{fleetv1beta1.CRPTrackingLabel: testCRPName},
+		},
+		Spec: fleetv1beta1.ResourceBindingSpec{
+			State:                        fleetv1beta1.BindingStateScheduled,
+			ResourceSnapshotName:         testResourceSnapshotName,
+			SchedulingPolicySnapshotName: testSchedulingPolicySnapshotName,
+			TargetCluster:                testTargetCluster,
+		},
+	}
+}
+
+func updateClusterResourceBindingStatusWithCondition(conditionType fleetv1beta1.ResourceBindingConditionType) {
+	crb := &fleetv1beta1.ClusterResourceBinding{}
+	By(fmt.Sprintf("Updating the clusterResourceBinding status - %s Successful", conditionType))
+	Expect(k8sClient.Get(ctx, types.NamespacedName{Name: testCRBName}, crb)).Should(Succeed(), "failed to get cluster resource binding")
+	crb.Status.Conditions = []metav1.Condition{
+		{
+			Type:               string(conditionType),
+			Status:             metav1.ConditionTrue,
+			Reason:             testReason,
+			LastTransitionTime: metav1.Now(),
+		},
+	}
+	Expect(k8sClient.Status().Update(ctx, crb)).Should(Succeed(), "failed to update cluster resource binding status")
```

Comment:
add more tests by updating status, eg, observedGeneration, reason, lastTransistionTime.

eg, when lastTransistionTime is changed but others are not changed.